gamemodes/terrortown/gamemode/client/cl_player_ext.lua

```diff
@@ -282,3 +290,105 @@ end
 function plymeta:GetRevivalReason()
 	return self.revivalReason or {}
 end
+
+---
+-- Sets a shared playersetting from the client on both server and client.
+-- @param string identifier The identifier of the shared setting
+-- @param any value The setting's value
+-- @realm client
+function plymeta:SetSettingOnServer(identifier, value)
+	if self.playerSettings[identifier] == value then return end
+
+	self.playerSettings[identifier] = value
+
+	net.Start("ttt2_set_player_setting")
+	net.WriteString(identifier)
+	net.WriteTable({value}, true) -- use table to support any data type
+	net.SendToServer()
+end
```

Comment:
I see what you mean. I'm not a huge fan of your suggestion though. Is this really a concern? This can only happen when the server has cheats enabled and a player wants to abuse this by sending lots of data

Reply:
ok, it might overcomplicate this whole ordeal, but what about the following:
only identifiers that are registered on the server and the client can be set, that way the only way to abuse is to overwrite them with a table, this is finite.

If that's not enough, one could go one step forward and only allow strings

Reply:
Another solution would be to use `GetInfoNum` and add a custom change detection (because you cannot attach a on change callback to a dsynced convar for some reason):

![image](https://github.com/TTT-2/TTT2/assets/13639408/7324995e-9f6e-4d12-841e-c96d3f2fb1e0)

Reply:
@saibotk could you take a look at this suggestion so I can finish this PR?

Reply:
Nah this not really nice and also not really a difference to the "only send strings" approach.

Maybe lets just use integers for this, so they can be used for 0 = false, 1 = true or could be used as some sort of index value for selects etc. 
I think we do not have use cases for strings or entire tables currently and in the near future.
Also limit the settings a user can set by having a serverside list of possible settings.

I think this would be an okayish solution to this.

I am not 100% sure how severe this really is, but you should never trust the client in anything. A malicious client does not care about sv_cheats and anyone can easily execute custom lua on the client. But id rather like to be a bit careful with such generic network functions though.

Maybe @Histalek has some opinion on this?

Reply:
I need floats though. Should I parse everything as a float then, even bools? or send everything as a string and define on the serverside the type of that variable?

Reply:
yes lets do that, then it is basically like convars